src/cunumeric/index/wrap_template.inl

```diff
@@ -60,17 +60,37 @@ struct WrapImpl {
     assert(volume_in != 0);
 #endif
 
-    WrapImplBody<KIND, DIM>()(out, pitches_out, out_rect, pitches_in, input_rect, dense);
+    if (args.has_input) {
+      auto in_rect = args.in.shape<1>();
+      auto in = args.in.read_accessor<int64_t, 1>(in_rect);  // input should be always integer type
+#ifdef DEBUG_CUNUMERIC
+      assert(in_rect == out_rect);
+#endif
+#ifndef LEGION_BOUNDS_CHECKS
+      dense = dense && in.accessor.is_dense_row_major(out_rect);
+#endif
+      WrapImplBody<KIND, DIM>()(out, pitches_out, out_rect, pitches_in, input_rect, dense, in);
+
+    } else {
+      bool tmp = false;
+      WrapImplBody<KIND, DIM>()(out, pitches_out, out_rect, pitches_in, input_rect, dense, tmp);
+    }  // else
   }
 };
 
 template <VariantKind KIND>
 static void wrap_template(TaskContext& context)
 {
-  auto shape = context.scalars()[0].value<DomainPoint>();
-  int dim    = shape.dim;
-  WrapArgs args{context.outputs()[0], shape};
-  dim_dispatch(dim, WrapImpl<KIND>{}, args);
+  auto shape     = context.scalars()[0].value<DomainPoint>();
+  int dim        = shape.dim;
+  bool has_input = context.scalars()[1].value<bool>();
+  if (has_input) {
+    WrapArgs args1{context.outputs()[0], shape, has_input, context.inputs()[0]};
```

Comment:
Can the two branches be combined in a single declaration?
```
WrapArgs args{context.outputs()[0], shape, has_input, has_input ? context.inputs()[0] : Array()};
```

Reply:
sure, thank you